### PR TITLE
Add support for system prompt on vertex

### PR DESCRIPTION
**Pull request by ImPedro29**
## Status
I've just made the code and some tests, in next days I'll start to work on bellow checklist. If you want this feature you can use  my branch

### PR Checklist

- [ ] Read the [Contributing documentation](https://github.com/tmc/langchaingo/blob/main/CONTRIBUTING.md).
- [ ] Read the [Code of conduct documentation](https://github.com/tmc/langchaingo/blob/main/CODE_OF_CONDUCT.md).
- [ ] Name your Pull Request title clearly, concisely, and prefixed with the name of the primarily affected package you changed according to [Good commit messages](https://go.dev/doc/contribute#commit_messages) (such as `memory: add interfaces for X, Y` or `util: add whizzbang helpers`).
- [ ] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `Fixes #123`).
- [ ] Describes the source of new concepts.
- [ ] References existing implementations as appropriate.
- [ ] Contains test coverage for new functions.
- [ ] Passes all [`golangci-lint`](https://golangci-lint.run/) checks.

### CODE REVIEW

**Comment by tmc** [comment]
@ImPedro29 is this ready for review? please mark as non-draft if so.

File: llms/googleai/vertex/vertex.go [hunk 1]
```diff
@@ -229,6 +231,25 @@ func convertContent(content llms.MessageContent) (*genai.Content, error) {
...
+	for _, part := range content.Parts {
+		prompt, ok := part.(llms.TextContent)
+		if !ok {
+			return nil, ErrSystemJustSupportsText
+		}
+		mergedSystemPrompt = append(mergedSystemPrompt, prompt.Text)
+	}
+
+	c.Role = RoleSystem
+	merged := genai.Text(strings.Join(mergedSystemPrompt, ".\n"))
```
Comment (wangjiancn) [inline llms/googleai/vertex/vertex.go:247]: Supporting multiple system prompts using splicing may have unexpected effects, so it is not recommended to modify them, and developers should be in control of their own system prompts

File: llms/googleai/vertex/vertex.go [hunk 2]
```diff
@@ -333,76 +367,94 @@ DoStream:
 	return convertCandidates([]*genai.Candidate{candidate})
 }
 
-// convertTools converts from a list of langchaingo tools to a list of genai
-// tools.
-func convertTools(tools []llms.Tool) ([]*genai.Tool, error) {
```
Comment (wangjiancn) [inline llms/googleai/vertex/vertex.go:338]: Why does this need modification? Debugging revealed no issues here, only facing the problem in https://github.com/tmc/langchaingo/pull/888. Please provide the next error message; we may have similar issues.

**Comment by ImPedro29** [comment]
Closing to follow repository rules. Fork bad used. I'll send a new PR soon

**Pull request closed without merging**